python/pyarrow/tests/test_udf.py

```diff
@@ -0,0 +1,483 @@
+def raise_func_fixture():
+    def raise_func(ctx):
+        raise ValueError("Test function with raise")
+    func_name = "test_raise"
+    doc = {
+        "summary": "test function with raise",
+        "description": "function with a raise"
+    }
+    pc.register_scalar_function(raise_func, func_name, doc,
+                                {}, pa.int64())
+    return raise_func, func_name
+
+
+def check_scalar_function(func_fixture,
+                          input,
+                          mock_udf_context,
+                          run_in_dataset=True):
+    function, name = func_fixture
+    expected_output = function(mock_udf_context, *input)
+    func = pc.get_function(name)
+    assert func.name == name
+
+    result = pc.call_function(name, input)
+
+    assert result == expected_output
+    if run_in_dataset:
+        field_names = [f'field{index}' for index, in_arr in input]
+        table = pa.Table.from_arrays(input, field_names)
+        dataset = ds.dataset(table)
+        func_args = [ds.field(field_name) for field_name in field_names]
+        result_table = dataset.to_table(
+            columns={'result': ds.field('')._call(name, func_args)})
+        assert result_table.column(0).chunks[0] == expected_output
+
+
+def test_scalar_udf_array_unary(unary_func_fixture, mock_udf_context):
+    check_scalar_function(unary_func_fixture,
+                          [
+                              pa.array([10, 20], pa.int64())
+                          ],
+                          mock_udf_context
+                          )
+
+
+def test_scalar_udf_array_binary(binary_func_fixture, mock_udf_context):
+    check_scalar_function(binary_func_fixture,
+                          [
+                              pa.array([10, 20], pa.int64()),
+                              pa.array([2, 4], pa.int64())
+                          ],
+                          mock_udf_context
+                          )
+
+
+def test_scalar_udf_array_ternary(ternary_func_fixture, mock_udf_context):
+    check_scalar_function(ternary_func_fixture,
+                          [
+                              pa.array([10, 20], pa.int64()),
+                              pa.array([2, 4], pa.int64()),
+                              pa.array([5, 10], pa.int64())
+                          ],
+                          mock_udf_context
+                          )
+
+
+def test_scalar_udf_array_varargs(varargs_func_fixture, mock_udf_context):
+    check_scalar_function(varargs_func_fixture,
+                          [
+                              pa.array([2, 3], pa.int64()),
+                              pa.array([10, 20], pa.int64()),
+                              pa.array([3, 7], pa.int64()),
+                              pa.array([20, 30], pa.int64()),
+                              pa.array([5, 10], pa.int64())
+                          ],
+                          mock_udf_context
+                          )
+
+
+def test_udf_input():
```

Comment:
Nit, but can you give more meaningful names to test functions? For example here: `test_registration_errors`.

Reply:
Sure. Added.